src/Toolkit/Toolkit/GeoViewController.cs

```diff
@@ -0,0 +1,279 @@
+            };
+            geoView.Loaded += _loadedListener.OnEvent;
+
+            _unloadedListener = new WeakEventListener<GeoViewController, GeoView, object?, LoadedEventArgs>(this, geoView)
+            {
+                OnEventAction = static (instance, source, eventArgs) => instance.GeoView_Unloaded((GeoView)source!),
+                OnDetachAction = static (instance, source, weakEventListener) => source.Unloaded -= weakEventListener.OnEvent,
+            };
+            geoView.Loaded += _loadedListener.OnEvent;
+            geoView.Unloaded += _unloadedListener.OnEvent;
+            OnGeoViewAttached(geoView);
+            if (geoView.IsLoaded)
+                GeoView_Loaded(geoView);
+        }
+
+        private void DetachFromGeoView(GeoView geoView)
+        {
+            var connectedView = ConnectedView;
+            if (connectedView != null && connectedView == geoView)
+            {
+                _loadedListener?.Detach();
+                _unloadedListener?.Detach();
+                if (connectedView.IsLoaded)
+                    GeoView_Unloaded(geoView);
+                OnGeoViewDetached(connectedView);
+                _geoViewWeak = null;
+            }
+        }
+
+        /// <summary>
+        /// Raised when the <see cref="GeoViewController"/> has been attached to a <see cref="GeoView"/>.
+        /// </summary>
+        /// <param name="geoView"></param>
+        protected virtual void OnGeoViewAttached(GeoView geoView)
+        {
+        }
+
+        /// <summary>
+        /// Raised when the <see cref="GeoViewController"/> has been detached from a <see cref="GeoView"/>.
+        /// </summary>
+        /// <param name="geoView"></param>
+        protected virtual void OnGeoViewDetached(GeoView geoView)
+        {
+        }
+
+        /// <summary>
+        /// Raised when the attached <see cref="GeoView"/> loads into the active view.
+        /// </summary>
+        /// <param name="geoView">GeoView that was loaded</param>
+        protected virtual void OnGeoViewLoaded(GeoView geoView)
+        {
+        }
+
+        /// <summary>
+        /// Raised when the attached <see cref="GeoView"/> unloads from the active view.
+        /// </summary>
+        /// <param name="geoView">GeoView that was unloaded</param>
+        protected virtual void OnGeoViewUnloaded(GeoView geoView)
+        {
+        }
+
+        private void GeoView_Loaded(GeoView geoView)
+        {
+            Debug.Assert(geoView == ConnectedView && ConnectedView is not null);
+            OnGeoViewLoaded(geoView);
+        }
+
+        private void GeoView_Unloaded(GeoView sender)
+        {
+            Debug.Assert(sender == ConnectedView && ConnectedView is not null);
+            OnGeoViewUnloaded(sender);
+        }
+
+        #region Viewpoints
+
+        /// <inheritdoc cref="GeoView.GetCurrentViewpoint(ViewpointType)"/>
```

Comment:
I didn't know we can just inherit docs across unrelated classes. Neat!

Reply:
It works for intellisense - however I don't think docfx likes it, so be careful.